proxy/round_tripper/proxy_round_tripper.go

```diff
@@ -163,9 +164,14 @@ func (rt *roundTripper) RoundTrip(originalRequest *http.Request) (*http.Response
 		trace.Reset()
 
 		if reqInfo.RouteServiceURL == nil {
+			if attempt > 1 {
+				if attempt > reqInfo.RoutePool.NumEndpoints() && reqInfo.RoutePool.LastUpdated().Equal(lastUpdated) {
```

Comment:
@maxmoehl wouldn't this prevent it from retrying already-tried endpoints even if MaxAttempts is set to a very large number?

Reply:
How? 😄 Once the timestamps differ it can no longer stop and if all instances fail it will loop until MaxAttempts is hit.